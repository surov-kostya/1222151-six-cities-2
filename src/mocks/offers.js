import {shape, number, string, arrayOf} from 'prop-types';

export const placeType = shape({
  id: number,
  name: string,
  imageSrc: string,
  contacts: shape({eMail: string}),
  gallerySrcs: arrayOf(string),
  mark: string,
  rating: number,
  parametres: shape({
    entire: string,
    bedrooms: number,
    adults: number
  }),
  price: number,
  features: arrayOf(string),
  host: shape({
    id: number,
    name: string,
    status: string
  }),
  hostsReview: arrayOf(string)
});

export const places = [
  {
    id: 0,
    name: `Beautiful && luxurious apartment at great location`,
    imageSrc: `img/apartment-01.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/map.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  },
  {
    id: 1,
    name: `Wood and stone place`,
    imageSrc: `img/apartment-02.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  },
  {
    id: 2,
    name: `Nice, cozy, warm big bed apartment`,
    imageSrc: `img/apartment-03.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    imageSrc: `img/room.jpg`,
    contacts: {
      eMail: `Oliver.conner@gmail.com`
    },
    gallerySrcs: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    mark: `Premium`,
    rating: 4.8,
    parametres: {
      entire: `Entire place`,
      bedrooms: 3,
      adults: 4
    },
    price: 120,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    host: {
      id: 100,
      name: `Angelina`,
      status: `Pro`
    },
    hostsReview: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ]
  }
];
