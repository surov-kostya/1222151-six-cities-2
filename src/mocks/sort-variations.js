export const variations = [
  {
    id: 0,
    name: `Popular`
  },
  {
    id: 1,
    name: `Price: low to high`
  },
  {
    id: 2,
    name: `Price: high to low`
  },
  {
    id: 3,
    name: `Top rated first`
  }
];
