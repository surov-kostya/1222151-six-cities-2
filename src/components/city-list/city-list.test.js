import React from 'react';
import renderer from 'react-test-renderer';
import CityList from './city-list';
import {cities} from '../../mocks/offers';

it(`City list renders correctly`, () => {
  const tree = renderer.create(<CityList
    cities={cities.map((city) => ({id: city.id, coords: city.coords, name: city.name}))}
    currentCityId={0}
    onChooseCity={() => {}}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
