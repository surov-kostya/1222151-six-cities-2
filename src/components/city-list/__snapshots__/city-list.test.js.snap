// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City list renders correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      onClick={[Function]}
    >
      <span>
        Paris
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__itemfalse"
      onClick={[Function]}
    >
      <span>
        Cologne
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__itemfalse"
      onClick={[Function]}
    >
      <span>
        Brussels
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__itemfalse"
      onClick={[Function]}
    >
      <span>
        Amsterdam
      </span>
    </a>
  </li>
</ul>
`;
